mlrun/datastore/store_resources.py

```diff
@@ -0,0 +1,128 @@
+# Copyright 2018 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# flake8: noqa  - this is until we take care of the F401 violations with respect to __all__ & sphinx
+
+import mlrun
+from mlrun.config import config
+from mlrun.utils.helpers import parse_function_uri
+from .v3io import v3io_path
+from ..utils import StorePrefix, parse_store_uri
+
+
+class ResourceCache:
+    """Resource cache for real-time pipeline/serving and storey"""
+
+    def __init__(self):
+        self._tabels = {}
+        self._resources = {}
+
+    def cache_table(self, uri, value, is_default=False):
+        """Cache storey Table objects"""
+        self._tabels[uri] = value
+        if is_default:
+            self._tabels["."] = value
+
+    def get_table(self, uri):
+        """get storey Table object by uri"""
+        try:
+            from storey import Table, Driver, V3ioDriver
+        except ImportError:
+            raise ImportError("storey package is not installed, use pip install storey")
+
+        if uri in self._tabels:
+            return self._tabels[uri]
+        if uri in [".", ""]:
+            self._tabels[uri] = Table("", Driver())
+            return self._tabels[uri]
+
+        if uri.startswith("v3io://") or uri.startswith("v3ios://"):
+            endpoint, uri = v3io_path(uri)
+            self._tabels[uri] = Table(uri, V3ioDriver(webapi=endpoint))
+            return self._tabels[uri]
+
+        # todo: map store:// uri's to Table objects
+
+        raise ValueError(f"table {uri} not found in cache")
+
+    def cache_resource(self, uri, value, default=False):
+        """cache store resource (artifact/feature-set/feature-vector)"""
+        self._resources[uri] = value
+        if default:
+            self._resources["."] = value
+
+    def get_resource(self, uri):
+        """get resource from cache by uri"""
+        return self._resources[uri]
+
+    def resource_getter(self, db=None, secrets=None):
+        """wraps get_store_resource with a simple object cache"""
+
+        def _get_store_resource(uri, use_cache=True):
+            if (uri == "." or use_cache) and uri in self._resources:
+                return self._resources[uri]
+            resource = get_store_resource(uri, db, secrets=secrets)
```

Comment:
Note that in `get_store_resource` you didn't add or documented the `use_cache` kwarg - so no one will know it has the option to not using the cache

Reply:
added, the only place its used is in the serving context (added to it dynamically and doc string may not hold)